jaggr-core/src/main/java/com/ibm/jaggr/core/cache/IGenericCache.java

```diff
@@ -81,4 +71,12 @@
 	 * @param aggregator The aggregator instance for the cache
 	 */
 	public void setAggregator(IAggregator aggregator);
+
+	/**
+	 * Returns a new, empty, instance of this cache object. Used by the cache manager when clearing
+	 * the aggregator cache.
+	 *
+	 * @return a new instance of this cache object
+	 */
+	public IGenericCache newInstance();

```

Comment:
Both changes (the cache listener and the `newInstance()` method) were added to allow aggregator extensions, like resource converters, to extend the cache by adding their own cache objects.  The cache objects are managed by the cache manager, but the semantics of the cache are determined by the implementation.  The cache object just needs to implement the `IGenericCache` interface.  

The cache listener is admittedly a bit of a hack and is needed because the cache manager is not initialized when the aggregator extensions are initialized.  I initially tried to change the order of initialization so that the cache manager would be initialized before the extensions, but this approach turned out to be rather problematic and the listener approach was the least disruptive solution.

As for the `IGenericCache::newInstance()` method, this used by the cache manager when clearing the aggregator cache.  The way that the aggregator cache is cleared is that the cache manager create new, empty cache objects, replaces the existing cache with the new cache in one atomic operation, and then lazily cleans up the old cache by calling the `clear()` methods on the old cache objects.  This avoids the need for synchronization when clearing the cache.  The `newInstance()` method is needed because the cache manager doesn't know how to create the cache objects that were added by extensions using the `ICache::putIfAbsent()` method, so it instead asks the cache objects to create new instances of themselves.